Makefile

```diff
@@ -79,82 +120,118 @@ docker-push:
 #####################
 
 # Run go linters
+.PHONY: go-lint
 go-lint: golangci-lint fmt vet generate
 	$(GOLANGCI_LINT) run --verbose
 
 # Run go fmt against code
+.PHONY: fmt
 fmt:
 	go fmt ./...
 
 # Run go vet against code
+.PHONY: vet
 vet: generate
 	go vet ./...
 
 # Check markdown files syntax
+.PHONY: md-lint
 md-lint: markdownlint
 	$(MARKDOWNLINT) \
 		-c "$(CURDIR)/.markdownlint.json" \
 		--ignore "$(CURDIR)/vendor" \
+		--ignore "$(CURDIR)/node_modules" \
 		"$(CURDIR)"
 
+docker-lint: hadolint
+	$(HADOLINT) Dockerfile
+
+make-lint: checkmake
+	$(CHECKMAKE) Makefile
+
 #####################
 #    Dev helpers    #
 #####################
 
 # Install CRDs into a cluster
-install: manifests
+.PHONY: install
+install: generate kustomize
 	$(KUSTOMIZE) build config/crd \
 		| kubectl apply --validate=false -f -
 
 # Uninstall CRDs from a cluster
-uninstall: manifests
+.PHONY: uninstall
+uninstall: generate kustomize
 	$(KUSTOMIZE) build config/crd \
 		| kubectl delete -f -
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: manifests
+.PHONY: deploy
+deploy: generate kustomize
 	cd config/manager && $(KUSTOMIZE) edit set image controller="$(IMG)"
 	$(KUSTOMIZE) build config/default \
 		| kubectl apply --validate=false -f -
 
-sample: gomplate
-	export \
-		LBAAS_DOMAIN=$$(kubectl get svc nginx-nginx-ingress-controller -o jsonpath='{.status.loadBalancer.ingress[0].hostname}') \
-		NOTARY_DOMAIN=$$(kubectl get svc nginx-nginx-ingress-controller-notary -o jsonpath='{.status.loadBalancer.ingress[0].hostname}') \
-		CORE_DATABASE_SECRET=$$(kubectl get secret core-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) \
-		CLAIR_DATABASE_SECRET=$$(kubectl get secret clair-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) \
-		NOTARY_SERVER_DATABASE_SECRET=$$(kubectl get secret notary-server-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) \
-		NOTARY_SIGNER_DATABASE_SECRET=$$(kubectl get secret notary-signer-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) ; \
-	kubectl kustomize config/samples \
-		| gomplate \
+# Deploy RBAC in the configured Kubernetes cluster in ~/.kube/config
+.PHONY: deploy-rbac
+deploy-rbac: generate kustomize
+	$(KUSTOMIZE) build config/rbac \
+		| kubectl apply --validate=false -f -
+
+.PHONY: sample
+sample: kustomize
```

Comment:
Rework this target to deploy a minimal usable harbor stack

Reply:
Done